src/constructs/numbers.md

```diff
@@ -2,6 +2,120 @@
 title: Numbers
 ---
 
-## On the Subject of Numbers
+Grain has a unified number type, called `Number`, which contains integer (whole) numbers, floating-point (decimal) numbers, and rational (fractional) numbers. Grain also has separate integer and float types for programs which require more precise typechecking.
 
-This documentation hasn't yet been migrated here (or written, to be honest). In the meanwhile, please check out the [Grain compiler README](https://github.com/grain-lang/grain/blob/master/README.md) for more information. Consider contributing by fleshing out this page yourself!
\ No newline at end of file
+```grain
+42 // an integer
+```
+
+```grain
+1.23 // a floating-point number
+```
+
+```grain
+1/3 // a rational number
+```
+
+## Number Syntax
+
+### Integers
+
+To make an integer more readable, underscores can be used throughout the number as a visual separator. The underscores do not affect the value of the number.
+
+```grain
+let oneBillion = 1_000_000_000
+```
+
+In addition to decimal numbers, integers can be written in hexadecimal, octal, and binary:
+
+```grain
+// hexadecimal
+let fourtyTwo = 0x2A // or 0x2a
+```
+
+```grain
+// octal
+let fourtyTwo = 0o52
+```
+
+```grain
+// binary
+let fourtyTwo = 0b101010
+```
+
+Of course, underscores can also be used with these other formats:
+
+```grain
+let dec0de = 0b1101_1110_1100_0000_1101_1110
+```
+
+### Floating-point Numbers
+
+Floating-point numbers can also be written using scientific notation. In this form, the "significand" part of the number is multiplied by 10 raised to the power of the provided exponent. For example,
```

Comment:
See first comment above. Weird to see "can also" when there is no content in this section. So move the normal example in here.

Reply:
Additionally, I don't know the word "significand", so we should probably explain it or use a more layman way of saying it.